typing/includemod.ml

```diff
@@ -795,28 +812,31 @@ and signature_components  ~in_eq ~loc old_env ~mark env subst
         | _ ->
             assert false
       in
-      let r =
+      let deep_modifications = !shape_modified in
+      let first =
         match item with
+        | Ok x ->
+            let runtime_coercions =
+              if present_at_runtime then [pos,x] else []
+            in
+            Sign_diff.{ empty with deep_modifications; runtime_coercions }
+        | Error { recoverable = true; error } ->
+            Sign_diff.{ empty with errors=[id,error]; deep_modifications }
         | Error { recoverable = false; error } ->
-            { runtime_coercions=[];
-              shape_map;
-              deep_modifications= !shape_modified;
-              errors=[id,error];
-              leftovers=rem
-            }
-        | Error { recoverable = true; _ } | Ok _ ->
-            signature_components ~in_eq ~loc old_env ~mark env subst
-              orig_shape shape_map rem
+            let errors =[id,error] in
+            Sign_diff.{ empty with deep_modifications; errors; leftovers=rem }
```

Comment:
if you allow me a minor nitpick: I think that `leftovers=rem` should come from `rest`, not `first`. (Nothing here, and `{ empty with leftovers=rem }` in the `if not continue` case of `rest`.)

Reply:
That's slightly cleared indeed. I will keep that in mind in subsequent updates.

Reply:
Well, since the change entry needed to be synced with trunk anyway, here is the last minute change.